bark/models/dynamic/single_track.hpp

```diff
@@ -148,19 +157,15 @@ inline double CalculateSteeringAngle(const SingleTrackModelPtr& model,
   double delta = f_state.angle + atan2(-gain * f_state.lat, vel);
 
   if (limit_steering) {
-    double delta_max_right =
-        std::min(model->GetSteeringAngleMax(),
-                 std::abs(std::atan2(model->GetLatAccelerationRightMax() *
-                                         model->GetWheelBase(),
-                                     vel * vel)));
-    double delta_max_left =
-        std::min(model->GetSteeringAngleMax(),
-                 std::abs(std::atan2(
-                     model->GetLatAccelerationLeftMax() * model->GetWheelBase(),
-                     vel * vel)));
-    double clamped_delta = std::min(delta, delta_max_left);
-    clamped_delta = std::max(clamped_delta, -delta_max_right);
-    return clamped_delta;
+    double wb = model->GetWheelBase();
+    double delta_max_acc = SteeringAngle(model->GetLatAccelerationMax(), vel, wb);
+    double delta_min_acc = SteeringAngle(model->GetLatAccelerationMin(), vel, wb);
+    VLOG(5) << "DeltaMaxAcc: " << delta_max_acc << ", DeltaMinAcc: " << delta_min_acc << ", LatAccMax: " << model->GetLatAccelerationMax() << ", LatAccMin: " << model->GetLatAccelerationMin();
+
+    double delta1 = boundValue(delta, -model->GetSteeringAngleMax(), model->GetSteeringAngleMax());
```

Comment:
should these two lines not be swapped?

Could potentially lead to the bug we have experienced